engine/access/rpc/backend/backend_transactions.go

```diff
@@ -220,14 +220,37 @@ func (b *backendTransactions) GetTransactionResult(
 		return nil, txErr
 	}
 
+	// derive status of the transaction once, assuming that it has not yet executed
```

Comment:
effecitvely call the `deriveTransactionStatus` twice - once with `executed=false` once with `true`